services/user.js

```diff
@@ -351,6 +351,22 @@ class User {
         return [];
     }
 
+    async isEmailAndIDPLoginPermitted(email, idp) {
+        const result = await this.userCollection.aggregate([
+            {
+                "$match": {
+                    email: email,
+                    IDP: idp,
+                    userStatus:{
+                        $ne: USER.STATUSES.ACTIVE
+                    }
+                }
+            },
+            {"$limit": 1} // return one
+        ]);
+        return result?.length === 0;
```

Comment:
I think we need to add a check here to verify that the result is not null and that result is an array. If it fails one of these checks then throw an error message that they query failed. If we have a situation where the query fails but no error is thrown then this function will return false and it will probably return an incorrect error message in the API response

Reply:
pushed for the update!